xarray/core/dataset.py

```diff
@@ -2627,6 +2627,107 @@ def stack(self, dimensions=None, **dimensions_kwargs):
             result = result._stack_once(dims, new_dim)
         return result
 
+    def to_stacked_array(self, new_dim, dims, variable_dim='variable'):
+        """Combine variables of differing dimensionality into a DataArray
+        without broadcasting.
+
+        This function is basically version of Dataset.to_array which does not
+        broadcast the variables.
+
+        Parameters
+        ----------
+        new_dim : str
+            Name of the new stacked coordinate
+        dims : Sequence[str]
+            Dimensions to be stacked. Not all variables in the dataset need to
+            have these dimensions.
+        variable_dim : str, optional
+            Name of the level in the MultiIndex object which corresponds to
+            the variables.
```

Comment:
I find this docstring confusing. What is "the Multindex object"? It is not obviously one of the arguments of the function.

Reply:
Ah yes. This is a bit tricky, I doubt that most users would every touch this argument. I added some more description to the "Returns" section which hopefully describes it better though.